OpenRA.Mods.Common/Widgets/TerrainTemplatePreviewWidget.cs

```diff
@@ -93,7 +93,7 @@ public override void Draw()
 					var u = gridType == MapGridType.Rectangular ? x : (x - y) / 2f;
 					var v = gridType == MapGridType.Rectangular ? y : (x + y) / 2f;
 					var pos = origin + scale * (new float2(u * ts.Width, (v - 0.5f * tileInfo.Height) * ts.Height) - 0.5f * sprite.Size);
-					Game.Renderer.SpriteRenderer.DrawSprite(sprite, pos, worldRenderer.Palette(Palette), size);
+					Game.Renderer.SpriteRenderer.DrawSprite(sprite, pos, worldRenderer.Palette(Template.Palette ?? Palette), size);
```

Comment:
This will force-override any palette overrides that the external code or yaml might set.  Could you set this up with a `GetPalette = t => t.Palette ?? Palette` function instead?